src/api/worker/offline/OfflineStorage.ts

```diff
@@ -415,6 +421,9 @@ AND NOT(${firstIdBigger("elementId", upper)})`
 		}
 
 		await this.deleteIn(MailTypeRef, listId, mailsToDelete.map(elementIdPart))
+
+		// We unlock access to the "ranges" db here. We lock it in order to prevent race conditions when accessing the "ranges" database.
+		await this.unlockRangesDbAccess(listId)
```

Comment:
There is no `finally` here, or what do you mean? :)

Reply:
I was also thinking to move it up, directly after the call to "updateRangeForList", as it otherwise unnecessary blocks the ranges database, even though we have already updated the range while deleting the emails may be taking some more time. What do you think?

Reply:
well that's the problem, there's no `finally` :sweat_smile: we should unlock unconditionally, even if something fails so we should wrap it in try/finally and unlock there.

I am not sure about the locking here, I think ~~it's safer to keep it this way, otherwise someone might add some emails to the range and we won't ever delete them?~~ no, way, that's not true, we would clean them up next time anyway. I guess we can move unlock earlier.

Reply:
Haha alright I see, I will add the try catch block, that makes absolutely sense :+1: